doc/developers.md

```diff
@@ -0,0 +1,73 @@
+These pages contain the C++ API [doxygen](https://doxygen.nl) documentation as well as some other useful information for developers.
+
+- 🐣 If you are a new user, you probably want to start with the project's [README.md](https://github.com/UCL/TDMS/blob/main/README.md) on github. There are installation and "getting started" instructions.
+- 🐛 If you've spotted a bug or want to request a feature, we have a short [CONTRIBUTING.md](https://github.com/UCL/TDMS/blob/main/CONTRIBUTING.md). Then please [send an issue through github](https://github.com/UCL/TDMS/issues).
+- If you want to contribute to the development (🚀!) and have read [CONTRIBUTING.md](https://github.com/UCL/TDMS/blob/main/CONTRIBUTING.md), you're in the right place.
+
+## What?
+
+The physics and method used by TDMS is described in [this pdf document](https://github.com/UCL/TDMS/blob/gh-doc/masterdoc.pdf) (the source is in [doc/latex](https://github.com/UCL/TDMS/blob/main/doc/latex)).
+In terms of the code, `tdms` is a C++ binary that can be installed in your path.
+We build with [CMake](https://cmake.org/) to be moderately platform-independent. We run CI builds and tests in Windows, Ubuntu, and MacOS.
+
+### Dependencies
+
+We depend on ...
+
+* [OpenMP](https://en.wikipedia.org/wiki/OpenMP) for parallelization.
+* [fftw](https://www.fftw.org/) to calculate numerical derivatives (numerical_derivative.h)
+* [spdlog](https://github.com/gabime/spdlog) for logging (this is installed automatically by a CMake [FetchContent](https://cmake.org/cmake/help/latest/module/FetchContent.html) if not found).
+* [MATLAB](https://www.mathworks.com/products/matlab.html) since we read in and write out MATLAB format files. We are actually thinking of removing this as a strict dependency ([#70](https://github.com/UCL/TDMS/issues/70)).
+
+### Code style and other admin
+
+We try to stick to [Google style C++](https://google.github.io/styleguide/cppguide.html) wherever practicable. <!-- And we run `clang-format` and `cppclean` linters as part of our CI. -->
+There is one exception: we prefer `#pragma once` at the top of each header (because it's one line rather than three and is almost the standard anyway).
+
+Please keep doxygen-style comments in the **headers**.
+```{.cpp}
+/**
+ * Add two numbers together
+ *
+ * @param a: real number
+ * @param b: real number
+ * @return sum of a and b
+ */
+double add(double a, double b);
+```
+And please document the header itself with at least:
+```{.cpp}
+/**
+ * @file filename.h
+ * 
+ * Describe the functions/classes in here.
+ */
+ #pragma once
+```
+You can quickly check the documentation for your changes look correct with:
+```{.sh}
+doxygen doc/Doxygen
+firefox html/index.html # or your web browser of choice
+```
+You should be able to find and read what you've changed.
+Don't worry about doxygen for the source files (although obviously please do write helpful comments there).
+
+### Compiling and debugging
+
```

Comment:
Add a line here with instructions to compile (e.g. `mkdir build; cd build; cmake ..; make` or whatever works for TDMS)?

Reply:
It's on the README. So I kinda didn't want to duplicate it. But you're probably right.

Reply:
Yeah, splitting bits of the same documentation over two different sources isn't ideal. Perhaps all the developer focused stuff could be stripped out of the README and into the new site, and README.md could be more user facing?